src/pages/index.js

```diff
@@ -1,129 +1,224 @@
-
   formList.forEach((formElement) => {
     const validator = new FormValidator(config, formElement);
     const formName = formElement.getAttribute("name");
-
     formValidators[formName] = validator;
     validator.enableValidation();
   });
 }
 
 enableValidation(validationConfig);
 
-function createCard(item) {
-  const newCard = new Card(item, "#card-template", handleImageClick);
-  return newCard.generateCard();
+function createCard(data) {
+  const card = new Card(
+    data,
+    userInfo.getUserId(),
+    "#card-template",
+    handleImageClick,
+    (cardId, cardElement) => {
+      return new Promise((resolve, reject) => {
+        confirmPopup.setSubmitAction(() => {
+          confirmPopup.renderLoading(true);
+          api
+            .deleteCard(cardId)
+            .then(() => {
+              cardElement.remove();
+              resolve();
+              confirmPopup.close();
+            })
+            .catch((err) => {
+              showError(err);
+              reject(err);
+            })
+            .finally(() => confirmPopup.renderLoading(false));
+        });
+        confirmPopup.open();
+      });
+    },
+    handleLikeClick
+  );
+
+  return card.generateCard();
 }
 
-openAddpopupButton.addEventListener("click", () => {
-  console.log("Add button clicked");
-  formValidators["add-form"].resetValidation(); 
-  addCardPopup.open();
-});
+function renderCard(data) {
+  const cardElement = createCard(data);
+  cardSection.addItem(cardElement);
+}
 
-document.addEventListener("DOMContentLoaded", () => {
-  cardSection.renderItems();
-});
+function handleImageClick(name, link) {
+  imagePopup.open({ name, link });
+}
+
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
+function handleProfileFormSubmit(formData) {
+  function makeRequest() {
+    return api.updateUserInfo(formData).then((res) => {
+      userInfo.setUserInfo({ name: res.name, about: res.about });
+    });
+  }
+  handleSubmit(makeRequest, editProfilePopup);
+}
+
+function handleAddCardFormSubmit(formData) {
+  function makeRequest() {
+    return api.addNewCard(formData).then((cardData) => {
+      renderCard(cardData);
+    });
+  }
+  handleSubmit(makeRequest, addCardPopup, "Creating...");
+}
+
+function handleAvatarFormSubmit(formData) {
+  function makeRequest() {
+    return api.updateAvatar({ avatar: formData.avatar }).then((res) => {
+      userInfo.setUserInfo({ avatar: res.avatar });
+    });
+  }
+  handleSubmit(makeRequest, avatarPopup);
+}
+
+function handleLikeClick(cardId, isLiked, cardInstance) {
+  const action = isLiked ? api.removeLike(cardId) : api.addLike(cardId);
+  action
+    .then((updatedCard) => {
+  
+    })
```

Comment:
* Likes don't work

` cardInstance.updateLikes(updatedCard.isLiked);` should be in `.then`, to toggle the like button

```js

function handleLikeClick(cardId, isLiked, cardInstance) {
  const action = isLiked ? api.removeLike(cardId) : api.addLike(cardId);
  action
    .then((updatedCard) => {
      cardInstance.updateLikes(updatedCard.isLiked);  // here
    })
    .catch((err) => {
      showError(err);
    });
}

```